src/fileformats/ocd_file_export.cpp

```diff
@@ -1949,9 +1949,15 @@ void OcdFileExport::setupTextSymbolSpecial(const TextSymbol* text_symbol, OcdTex
 	ocd_text_special.line_below_offset = decltype(ocd_text_special.line_below_offset)(convertSize(qRound(1000 * text_symbol->getLineBelowDistance())));
 	
 	ocd_text_special.num_tabs = text_symbol->getNumCustomTabs();
-	auto last_tab = std::min(ocd_text_special.num_tabs, decltype(ocd_text_special.num_tabs)(std::extent<typename std::remove_pointer<decltype(ocd_text_special.tab_pos)>::type>::value));
-	for (auto i = 0u; i < last_tab; ++i)
-		ocd_text_special.tab_pos[i] = convertSize(text_symbol->getCustomTab(i));
+	auto max_tabs = decltype(ocd_text_special.num_tabs)(std::extent<typename std::remove_pointer<decltype(ocd_text_special.tab_pos)>::type>::value);
+	if (ocd_text_special.num_tabs > max_tabs)
+	{
+		ocd_text_special.num_tabs = max_tabs;
+		addWarning(::OpenOrienteering::OcdFileExport::tr("In text symbol %1: exporting only %2 custom tabulator positions")
+		           .arg(text_symbol->getPlainTextName()).arg(max_tabs));
+	}
```

Comment:
I was wondering why it looks so complicated, why I have difficulties to understand the type conversion implications.
As a general pattern with potential value domain mismatches, I would probably approach it like this:
~~~C++
auto const num_tabs = text_symbol->getNumCustomTabs();  //  original type retained: int
auto const max_tabs = std::extent<typename std::remove_pointer<decltype(ocd_text_special.tab_pos)>::type>::value);  // original type retained: std::size_t
if (num_tabs <= max_tabs)  // Safe enough here? Read below...
{
     	ocd_text_special.num_tabs = num_tabs  //  safe type conversion now
}
else
{
     	ocd_text_special.num_tabs = max_tabs  //  safe type conversion by data structure design
        addWarning(...)
}
for (...)
~~~

Pitfall:
The problem with `num_tabs <= max_tabs` is that you normally shouldn't mix signed and unsigned types: You may end up with negative values from the signed type being reinterpreted as (large) positive values for the unsigned type. (Cf. [example for Arithmetic comparison operators](https://en.cppreference.com/w/cpp/language/operator_comparison).)
Here, it shouldn't be an issue because both variables should only contain small non-negative values.
But there might still be a compiler warning. To resolve this issue, I would probably just wrap the `std::extent(...)` with `int(...)`.

Reply:
> The problem with num_tabs <= max_tabs is that you normally shouldn't mix signed and unsigned types: You may end up with negative values from the signed type being reinterpreted as (large) positive values for the unsigned type. (Cf. [example for Arithmetic comparison operators](https://en.cppreference.com/w/cpp/language/operator_comparison).)

I'm glad that we repeat the C-language basics here for the new contributors. Yes, converting unsigned types to signed ones, or vice versa, and then attempting comparison or arithmetic on them leads to surprising effects. The reason, in my opinion, is that C-is a very low-level language that exposes the CPU features to the programmer and assumes that the programmer understands what he/she is working with. Rude people tend to say that C is not a programming language but rather a macro assembler. I consider this statement illustrative of the situation. 

We got to keep in mind that C++ maintains this tradition. Thankfully, the C++ creators try to give us tools to avoid such pitfalls, for example through the concept of narrowing conversion.

> Here, it shouldn't be an issue because both variables should only contain small non-negative values.

Ack. `num_tabs` will likely be non-negative. It would require a bit of effort (i.e. create over 65k custom tab positions) to make it overflow. `max_tabs` is fixed at 32.

> But there might still be a compiler warning. To resolve this issue, I would probably just wrap the std::extent(...) with int(...).

If we really had to fix the (almost-nonexistent) issue, how about adjusting the return type of `TextSymbol::getNumCustomTabs()`? I can see that the first narrowing conversion is happening right there.

I also suggest that we move the discussion elsewhere, say into the Discussions section or into a private developer-only thread to avoid scaring away our new contributors and prevent spamming the inboxes of people who are watching the GH project.

Reply:
> > But there might still be a compiler warning. To resolve this issue, I would probably just wrap the std::extent(...) with int(...).
> 
> If we really had to fix the (almost-nonexistent) issue, how about adjusting the return type of `TextSymbol::getNumCustomTabs()`? I can see that the first narrowing conversion is happening right there.

I prefer to avoid unsigned integers.

> I also suggest that we move the discussion elsewhere, say into the Discussions section or into a private developer-only thread to avoid scaring away our new contributors and prevent spamming the inboxes of people who are watching the GH project.

I don't agree here. My comments are for this PR and for @dl3sdo. I would try to approach other contributor's differently.
You are effectively asking that such PRs would need to be made to my personal fork instead.

(FTR this is only part of my communication with @dl3sdo.)

Reply:
> I don't agree here. My comments are for this PR and for @dl3sdo. I would try to approach other contributor's differently. You are effectively asking that such PRs would need to be made to my personal fork instead.

I was perhaps unclear in my message. I meant that there are 48 people watching this thread and only five of them contributed code in the last year. Given that this discussion shifted from the immediate bug fix towards clarification on the coding style, I thought we could stop spamming the remaining 43 people's inboxes with a topic that is truly developer-only.

> (FTR this is only part of my communication with @dl3sdo.)

That looks like another reason for taking the conversation aside. And if the message was intended only for Matthias, it truly helps to start the text with a salutation.

I'll refrain from further comments here. From my POV, Matthias did a good job here and the .ocd custom tab issue is resolved. Thanks!